shared/metrics/src/main/java/io/pravega/shared/metrics/StatsProviderImpl.java

```diff
@@ -66,84 +52,93 @@ public void start() {
-            }
-            outdir.mkdirs();
-            log.info("Configuring stats with csv output to directory [{}]", outdir.getAbsolutePath());
-            reporters.add(CsvReporter.forRegistry(getMetrics())
-                          .convertRatesTo(TimeUnit.SECONDS)
-                          .convertDurationsTo(TimeUnit.MILLISECONDS)
-                          .build(outdir));
-        }
         if (conf.isEnableStatsdReporter()) {
             log.info("Configuring stats with statsD at {}:{}", conf.getStatsDHost(), conf.getStatsDPort());
-            reporters.add(StatsDReporter.forRegistry(getMetrics())
-                                        .prefixedWith(conf.getMetricsPrefix())
-                                        .build(conf.getStatsDHost(), conf.getStatsDPort()));
-        }
-        if (conf.isEnableGraphiteReporter()) {
-            log.info("Configuring stats with graphite at {}:{}", conf.getGraphiteHost(), conf.getGraphitePort());
-            final Graphite graphite = new Graphite(new InetSocketAddress(conf.getGraphiteHost(), conf.getGraphitePort()));
-            reporters.add(GraphiteReporter.forRegistry(getMetrics())
-                .prefixedWith(conf.getMetricsPrefix())
-                .convertRatesTo(TimeUnit.SECONDS)
-                .convertDurationsTo(TimeUnit.MILLISECONDS)
-                .filter(MetricFilter.ALL)
-                .build(graphite));
-        }
-        if (conf.isEnableJMXReporter()) {
-            log.info("Configuring stats with jmx {}", conf.getJmxDomain());
-            final JmxReporter jmx = JmxReporter.forRegistry(getMetrics())
-                .inDomain(conf.getJmxDomain())
-                .convertRatesTo(TimeUnit.SECONDS)
-                .convertDurationsTo(TimeUnit.MILLISECONDS)
-                .build();
-            jmx.start();
-        }
-        if (conf.isEnableGangliaReporter()) {
-            try {
-                log.info("Configuring stats with ganglia at {}:{}", conf.getGangliaHost(), conf.getGangliaPort());
-                final GMetric ganglia = new GMetric(conf.getGangliaHost(), conf.getGangliaPort(), GMetric.UDPAddressingMode.MULTICAST, 1);
-                reporters.add(GangliaReporter.forRegistry(getMetrics())
-                    .prefixedWith(conf.getMetricsPrefix())
-                    .convertRatesTo(TimeUnit.SECONDS)
-                    .convertDurationsTo(TimeUnit.MILLISECONDS)
-                    .build(ganglia));
-            } catch (IOException e) {
-                log.warn("ganglia create failure: {}", e);
-            }
-        }
-        if (conf.isEnableConsoleReporter()) {
-            log.info("Configuring console reporter");
-            reporters.add(ConsoleReporter.forRegistry(getMetrics())
-                .convertRatesTo(TimeUnit.SECONDS)
-                .convertDurationsTo(TimeUnit.MILLISECONDS)
-                .build());
+            StatsdConfig config = new StatsdConfig() {
```

Comment:
Please make this a separate, private static class. You've inlined 2 such classes in one method, making it much harder to read. 


Same below.